bin/cli.js

```diff
@@ -147,18 +148,38 @@ const setRequestHeaders = () => {
   addUserAgentHeader('HubSpot CLI', pkg.version);
 };
 
-const NO_CONFIG_VALIDATION = {
-  init: { skip: true },
-  auth: { skip: true },
+const shouldTargetCommand = (options, commandMap) => {
+  const checkCommand = (command, map) => {
+    const mainCommand = command;
+
+    if (map[mainCommand]) {
+      if (map[mainCommand].target && options._.includes(mainCommand)) {
+        return false;
+      }
+      const subCommands = map[mainCommand].subCommands || {};
+      for (const subCommand in subCommands) {
+        if (!checkCommand(subCommand, subCommands)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  };
+
+  return options._.every(command => checkCommand(command, commandMap));
```

Comment:
We might be able to simplify this a little bit. From what I can tell this seems to work in most cases, but it gets there in a slightly roundabout way.

Kicking this off using `every()` means that you'll call `checkCommand()` for every command that gets passed in. So running `hs accounts remove` would result in us calling `checkCommand()` for `accounts` and then also for `remove`, which I'm pretty sure isn't necessary.

The logic flow currently looks something like this:
1. `checkCommand('accounts', SKIP_ACCOUNT_VALIDATION)`
2. The first if check skips because `SKIP_ACCOUNT_VALIDATION['accounts'].target` is false
3. We get subcommands and then iterate over them and recursively call checkCommand for each one like this: `checkCommand('clean', SKIP_ACCOUNT_VALIDATION['accounts'].subCommands)`
4. The first if check for "clean" would skip  because `SKIP_ACCOUNT_VALIDATION['clean'].target` is true but options._ doesn't include"clean". Eventually this returns true
5. We do this for "list", and "ls", and then we eventually get to "remove" where we return false
6.  This exits the for-in loop and returns false
7. This aborts the first `options._.every()` call because the first check returned `false`
8.  `shouldTargetCommand ` returns false

**A few takeaways:**
- The outcome would be equivalent here if you were to kick this off using `return checkCommand()` instead of using `every()`
- We'd be recursively calling `checkCommand()` for "clean", then "list", then "ls", and THEN "remove". In reality, all we care about is "remove" because that's the command that the user ran. If we had a third layer of nesting (like hs accounts list account) then we'd recur again before getting to "remove"

It looks like we're deciding to recursively call `checkCommand` based on the presence of `map[mainCommand].subCommands`, but this gets much simpler if we make the decision based off of the length of `options._`. If we make our decisions based off of the command that the user ran, it prevents us from doing extra cycles and makes the logic easier to follow. For example:
1. call `checkCommand(options, commandMap)`
2. grab `const currentCommand = options._[0]`
3. Check if `currentCommand` should be targeted or has subcommands. If there's no entry for it, return false. If it's targeted, then return true. If it isn't targeted but has subcommands, call `return checkCommand(options._.slice(1), commandMap[currentCommand])`

The benefit of doing it this way is that there aren't any unnecessary iterations because at most we call `checkCommand` once per command that the user entered.

The naming in here is also a confusing to me because the method is called `shouldTargetCommand`, and the value for `accounts.clean` is `{ target: true }`, but then we return false when the user runs `hs accounts clean`. If target is true, I'd personally expect `shouldTargetCommand` to return true for that command.

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1310/commits/7fd67f3b76f8a0d7570611193f6e301d600ce7f8